remix/app/routes/_app.newsletter.create.tsx

```diff
@@ -0,0 +1,55 @@
+import { ActionArgs, json } from '@remix-run/node';
+import { Form } from '@remix-run/react';
+import { withZod } from '@remix-validated-form/with-zod';
+import { validationError } from 'remix-validated-form';
+import { z } from 'zod';
+
+import appHandler from '../lib/handler/app.handler';
+import NewsletterRepository from '../repositories/newsletter.server';
+
+const validator = withZod(
+  z.object({
+    name: z.string().min(1, { message: 'Name is required' }),
+    content: z.string().min(1, { message: 'Content is required' }),
+  })
+);
+
+export async function action({ request }: ActionArgs) {
+  return appHandler(request, async (user) => {
+    const result = await validator.validate(await request.formData());
+
+    if (result.error) {
+      return validationError(result.error);
+    }
+
+    const newsletterData = {
+      ...result.data,
+      userId: user.id,
+    };
+
+    const newsletter = await NewsletterRepository.create({
+      data: newsletterData,
+    });
+
+    return json({ ...newsletter });
+  });
+}
+
+export default function Index() {
+  return (
+    <>
+      <h1>Create Newsletter page </h1>
+      <Form method="POST">
+        <label>Name</label>
+        <br />
+        <input type="text" name="name" placeholder="name" />
```

Comment:
Minor comment on this form structure as it is not ideal to repeat `<br>` all the time. Prefer wrapping each form attribute into a `div` and setting display property on the `label` and `input` tags:

```
<Form method="POST">
  <div class="form-input">
  	<label for"name">Name</label>
  	<input type="text" name="name" placeholder="Name" />
  </div>
```

Reply:
For the reason, that I repeat the `<br />` because I want to the form to look useable, as this is a Backend PR I avoid using any styling for now.

[#120] In this PR I have create a input and textarea components that warped by `div` and have display property.

Resolved remove `<br>` and wrap by `<div>` :  [2f66680](https://github.com/nimblehq/js-framework-benchmark/pull/112/commits/2f666801102f0d29110d5774ae5761f1f655199f)